### PR TITLE
feat/a ton of react examples

### PR DESCRIPTION
**Pull request by brunos3d**
This PR introduces a set of samples using different versions and approaches with React.

- [different-react-versions-16-17](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/different-react-versions-16-17)
- [different-react-versions-16-18](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/different-react-versions-16-18)
- [different-react-versions-16-17-typescript](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/different-react-versions-16-17-typescript)
- [different-react-versions-16-18-typescript](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/different-react-versions-16-18-typescript)
- [react-18-ssr](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/react-18-ssr)
- [react-16-17-18-ssr](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/react-16-17-18-ssr)
- [react-18-server-2-server](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/react-18-server-2-server)
- [loadable-react-18](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/loadable-react-18)
- [loadable-react-16](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/loadable-react-16)
- [react-18-code-splitting](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/react-18-code-splitting)
- [apollo-client](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/apollo-client)
- [styled-components](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/styled-components)

### CODE REVIEW

**Review by PeaceDate — CHANGES_REQUESTED** [review]
Could you please add suitable names for package.jsons: 
For example `appolo-client` sample
1. Root package.json name (react-18-code-splitting -> apollo-client) sample name for root package
2. For Nested app1 and app2 (apollo-client_app1, apollo-client_app2)

Please check all samples added samples

Cause at now you have an error when we try to find changed samples (workspaces) 
Screenshot 2023-02-28 at 11 07 26


To check yourself you can run `node checkChangedWorkspaces.js` script. The output need to be as an Array of changed samples (workspaces)

Thank you

**Comment by PeaceDate** [comment]
Also check samples in description
[loadable](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/loadable)
[react-spectrum](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/react-spectrum)
[loadable-react-spectrum-](https://github.com/brunos3d/module-federation-examples/tree/feat/a-ton-of-react-examples/loadable-react-spectrum-)

They returns 404 

In addition, I think there is `loadable-react-spectrum-` smth missing in the name

**Comment by brunos3d** [comment]
@PeaceDate Oh, thanks for pointing it, it wasn't to be there (at least not yet) :smile:

**Comment by PeaceDate** [comment]
@brunos3d Update fork please) 
Screenshot 2023-03-01 at 18 20 42

**Comment by brunos3d** [comment]
@PeaceDate Done :rocket:

File: react-16-17-18-ssr-with-adapters/package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (PeaceDate) [inline react-16-17-18-ssr-with-adapters/package.json]: Get an error during `yarn install` Error: Couldn't find package "react-version-adapter@0.1.0" required by "@react-ssr/remote2@0.0.0" on the "npm" registry.

**Review by PeaceDate — CHANGES_REQUESTED** [review]

File: react-16-17-18-ssr-with-adapters/package.json (hunk 1 above)
Comment (brunos3d) [reply react-16-17-18-ssr-with-adapters/package.json]: Fixed :rocket:

**Review by PeaceDate — APPROVED** [review]

**Review by CaioRolla — APPROVED** [review]

**Pull request merged**